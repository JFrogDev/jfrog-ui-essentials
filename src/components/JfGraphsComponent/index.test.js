import {shallowMount} from '@vue/test-utils';
import JfGraphsComponent from './index.vue';

describe('JfGraphsComponent', () => {

    it('should match the snapshot', () => {
        jest.useFakeTimers();
        const wrapper = shallowMount(JfGraphsComponent, {
            propsData: {
                options: {
                    data: {
                        type: "bar",
                        columns: [
                            ["data1", 30, 200, 100, 170, 150, 250],
                            ["data2", 130, 100, 140, 35, 110, 50]
                        ]
                    }
                }
            },
            mocks: {
            }
        });
        // jest.runAllTimers();
        expect(wrapper.vm.$el).toMatchSnapshot()
    });

    it('Contains an element to render graph', () => {
        const wrapper = shallowMount(JfGraphsComponent, {
            propsData: {}
        });
        expect(wrapper.find({ ref: 'chart' }).exists()).toBe(true);
    });

    it('contains a prop options', () => {
        const wrapper = shallowMount(JfGraphsComponent, {
            propsData: {
                options: 'chart-data'
            }
        });
        expect(wrapper.props().options).toBe('chart-data');
    });

});
