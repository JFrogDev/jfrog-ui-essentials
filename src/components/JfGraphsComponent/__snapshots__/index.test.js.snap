// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JfGraphsComponent should match the snapshot 1`] = `<div />`;
